frontend/public/components/utils/webhooks.tsx

```diff
@@ -55,8 +55,35 @@ export const WebhookTriggers: React.FC<WebhookTriggersProps> = (props) => {
     namespace,
   });
   const tableColumnClasses = getTableColumnClasses(canGetSecret);
-  const webhooks = _.filter(triggers, (trigger) => webhookTriggers.has(trigger.type));
-  if (_.isEmpty(webhooks)) {
+  const [webhookSecrets, setWebhookSecrets] = React.useState<K8sResourceKind[]>([]);
+  const [isLoaded, setIsLoaded] = React.useState<boolean>(false);
+
+  const webhooks: WebhookTrigger[] = _.filter(triggers, (trigger) =>
+    webhookTriggers.has(trigger.type),
+  );
+  const webhookNames: string[] = webhooks.map(
+    (webhook: WebhookTrigger): string => {
+      const triggerProperty = getTriggerProperty(webhook);
+      return _.get(webhook, [triggerProperty, 'secretReference', 'name']);
+    },
+  );
+
+  React.useEffect(() => {
+    Promise.all(
+      webhookNames.map(
+        (webhookName: string): Promise<K8sResourceKind> => {
+          return k8sGet(SecretModel, webhookName, namespace).catch(() => {
```

Comment:
Had to add the `.catch()` block since if there is an error retrieving the secret (eg. user manually edits the BC with a secret that does not exists...) we need to catch the error and return `null` , instead of uncaught error in promise